File: Solutions/0998/0998.py
# -*- coding:utf-8 -*-
# Author: washing
# DateTime: 2022/08/30 15:11
# File: 0998.py
# Desc: 


# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def insertIntoMaxTree(self, root: Optional[TreeNode], val: int) -> Optional[TreeNode]:
        t = TreeNode(val)
        if root.val < val:
            t.left = root
            return t
        cur = root
        while cur:
            if not cur.right:
                cur.right = t
                break
            elif cur.right.val < val:
                t.left = cur.right
                cur.right = t
                break
            cur = cur.right
        return root

File: Solutions/0998/0998.c
/***************************************************************************************************
 * Copyright: washing
 * FileName: 0998.c
 * Author: washing
 * Version: 1.0
 * Date: 2022-8-30 15:23:03
***************************************************************************************************/



/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */


struct TreeNode* insertIntoMaxTree(struct TreeNode* root, int val){
    struct TreeNode* t = (struct TreeNode *)malloc(sizeof(struct TreeNode));
    t->val = val;
    t->left = NULL;
    t->right = NULL;
    if (root->val < val) {
        t->left = root;
        return t;
    }
    struct TreeNode* cur = root;
    while (cur) {
        if (!cur->right) {
            cur->right = t;
            break;
        } else if (cur->right->val < val) {
            t->left = cur->right;
            cur->right = t;
            break;
        }
        cur = cur->right;
    }
    return root;
}
